api/src/main/java/org/openmrs/Obs.java

```diff
@@ -1190,6 +1252,31 @@ public void setFormField(String namespace, String formFieldPath) {
 			throw new APIException("Obs.namespaceAndPathNotContainSeparator", (Object[]) null);
 		}
 		
+		markAsDirty(this.formNamespaceAndPath, nsAndPathTemp);
 		formNamespaceAndPath = nsAndPathTemp;
 	}
+	
+	/**
+	 * Returns true if any change has been made to an Obs instance. In general, the only time
+	 * isDirty() is going to return false is when a new Obs has just been instantiated or loaded
+	 * from the database and no method that modifies it internally has been invoked.
+	 *
+	 * @return true if not changed otherwise false
+	 * @since 2.0
+	 * @should return false when no change has been made
+	 * @should return true when any immutable field has been changed
+	 * @should return false when only mutable fields are changed
+	 * @should return true when an immutable field is changed from a null to a non null value
+	 * @should return true when an immutable field is changed from a non null to a null value
+	 */
+	public boolean isDirty() {
+		return dirty;
+	}
+	
+	private void markAsDirty(Object oldValue, Object newValue) {
+		//Should we ignore the case for Strings?
```

Comment:
@wluyima do you know why bamboo is failing with this commit?
It passes locally and on travis.